github/pulls.go

```diff
@@ -248,19 +248,29 @@ type PullRequestMergeResult struct {
 	Message *string `json:"message,omitempty"`
 }
 
+type PullRequestOptions struct {
+	Squash bool
+}
+
 type pullRequestMergeRequest struct {
 	CommitMessage *string `json:"commit_message"`
+	Squash        *bool   `json:"squash,omitempty"`
 }
 
 // Merge a pull request (Merge Button™).
 //
 // GitHub API docs: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-buttontrade
-func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string) (*PullRequestMergeResult, *Response, error) {
+func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string, options *PullRequestOptions) (*PullRequestMergeResult, *Response, error) {
 	u := fmt.Sprintf("repos/%v/%v/pulls/%d/merge", owner, repo, number)
 
-	req, err := s.client.NewRequest("PUT", u, &pullRequestMergeRequest{
-		CommitMessage: &commitMessage,
-	})
+	pullRequestBody := &pullRequestMergeRequest{CommitMessage: &commitMessage}
+	if options != nil {
+		pullRequestBody.Squash = &options.Squash
+	}
+	req, err := s.client.NewRequest("PUT", u, pullRequestBody)
+
+	// TODO: This header will be unnecessary when the API is no longer in preview.
+	req.Header.Add("Accept", mediaTypeSquashPreview)
```

Comment:
> At quick glance, I still don't see why this method has multiple Accept headers.

I see it now. /cc @willnorris It's because of this line, it does `req.Header.Add("Accept", ...Preview)` here, instead of what's done everywhere else which is `req.Header.Set("Accept", ...Preview)`, e.g., see [here](https://github.com/gsquire/go-github/blob/09a37d57ac99a8938444e44375db8660c2184696/github/reactions.go#L167).

`NewRequest` adds the "Accept" header value to `mediaTypeV3` [here](https://github.com/google/go-github/blob/c2beba44cffbb17740cc3ad8d54f2b303060027b/github/github.go#L209).

@gsquire, if you change this line to be `req.Header.Add` to be consistent with how preview accept headers are set everywhere else, this should get rid of the need for changing `testHeader`.

@willnorris, it'll be good to confirm how multiple custom media types in the `Accept` header are supposed to overlap, and whether the current behavior of always having at most one `Accept` header value is correct (I think it is, but I haven't confirmed).